documentation/APIs/core-models.md

```diff
@@ -0,0 +1,581 @@
+        </blockquote>
+      </details>
+    </td>
+  </tr>
+  <tr>
+    <td>Latest Finalized Slot</td>
+    <td>int64</td>
+    <td>The slot index of latest finalized slot.</td>
+  </tr>
+  <tr>
+    <td>Payload Length</td>
+    <td>uint32</td>
+    <td>The length of the following payload in bytes. A length of 0 means no payload will be attached.</td>
+  </tr>
+  <tr>
+    <td valign="top">Payload <code>oneOf</code></td>
+    <td colspan="2">
+      <details open="true">
+        <summary>Generic Payload</summary>
+        <table>
+          <tr>
+            <th>Name</th>
+            <th>Type</th>
+            <th>Description</th>
+          </tr>
+          <tr>
+            <td>Payload Type</td>
+            <td>uint32</td>
+            <td>
+              The type of the payload. It will instruct the node how to parse the fields that follow.
+            </td>
+          </tr>
+          <tr>
+            <td>Payload Bytes</td>
+            <td>ByteArray</td>
+            <td>A sequence of serialized bytes, where the structure depends on `Payload Type`.</td>
+          </tr>
+        </table>
+      </details>
+    </td>
+  </tr>
+  <tr>
+    <td valign="top">Signature <code>oneOf</code></td>
+    <td colspan="2">
+      <details open="true">
+        <summary>Generic Signature</summary>
+        <table>
+          <tr>
+            <th>Name</th>
+            <th>Type</th>
+            <th>Description</th>
+          </tr>
+          <tr>
+            <td>Signature Type</td>
+            <td>uint8</td>
+            <td>
+              The type of the signature. It will instruct the node how to parse the fields that follow.
+            </td>
+          </tr>
+          <tr>
+            <td>Signature Bytes</td>
+            <td>ByteArray</td>
+            <td>A sequence of serialized bytes, where the structure depends on `Signature Type`.</td>
+          </tr>
+        </table>
+      </details>
+    </td>
+  </tr>
+  <tr>
+    <td>Nonce</td>
+    <td>uint64</td>
+    <td>The nonce which lets this block fulfill the PoW requirement.</td>
+  </tr>
+</table>
+
+
+### Block ID
+Block ID denotes an identifier of a block, with type `ArrayBytes[40]`. It is calculated as the following steps, using BLAKE2b-256 hash function:
+
+* `content` is the serialized block **without** signature and nonce.
+* `slot_index` is the slot index of the `Issuing Time` of the block.
+* `signatureBytes` is the serialized Ed25519 signature.
+
+Calculation:
+1. `content_hash` = hash(`content`)
+2. `id` = hash(Concat(`content_hash`, signatureBytes))
+3. And finally, `BlockID` = Concat(`id`, `slot_index`) 
+
+The string format of Block ID is hexadecimal encoding of Block ID with `0x` prefix.
+
+### Block Syntactic Validation Rules
+
+* Block size must not exceeds `MaxBlockSize` bytes, currently is `32768`.
+* `Protocol Version` must match the `Protocol Version` config parameter of the node.
+* Must have at least **1** `Strong Parents`.
```

Comment:
Just to be sure, this means we can have no weak and no shallow like parents? To disambiguate, I'd give an actual allowed range for each of them.